google/datalab/bigquery/_table.py

```diff
@@ -595,20 +598,24 @@ def to_dataframe(self, start_row=0, max_rows=None):
     Returns:
       A Pandas dataframe containing the table data.
     """
-    fetcher = self._get_row_fetcher(start_row=start_row, max_rows=max_rows)
+    fetcher = self._get_row_fetcher(start_row=start_row,
+                                    max_rows=max_rows,
+                                    page_size=self._MAX_PAGE_SIZE)
     count = 0
     page_token = None
-    df = None
+
+    # Collect results of page fetcher in separate dataframe objects, then
+    # concatenate them to reduce the amount of copying
+    df_list = []
+
     while True:
       page_rows, page_token = fetcher(page_token, count)
       if len(page_rows):
         count += len(page_rows)
-        if df is None:
-          df = pandas.DataFrame.from_records(page_rows)
-        else:
-          df = df.append(page_rows, ignore_index=True)
+        df_list.append(pandas.DataFrame.from_records(page_rows))
```

Comment:
I experimented with this a little, but I'm not seeing any speedup. It looks like creating a dataframe out of a list is a cheap operation, and extending large lists might sometimes result in copying so that might be offsetting any benefit from using one big list.